lib/customValidators.js

```diff
@@ -235,6 +252,25 @@ function isSrvrSecProperArray(schemaType, specialSecTypes, secObj, secName) {
   return true;
 }
 
+/**
+ * Validates if given server security is not an empty array when security type requires it
+ * @private
+ * @param  {String} schemaType security type, like httpApiKey or userPassword
+ * @param  {String[]} specialSecTypes list of special types that do not have to be an empty array
+ * @param  {Object} secObj server security object
+ * @param  {String} secName name os server security object
+ * @returns {String[]} there are 2 elements in array, index 0 is the name of the security schema object and index 1 is it's type
+ */
+function hasSrvrSecScopes(schemaType, specialSecTypes, secObj, secName) {
```

Comment:
I decided to have separate function instead of trying to mix it with `isSrvrSecProperArray` because code seems clearer with such an approach, and we can also nicely throw specific meaningful error